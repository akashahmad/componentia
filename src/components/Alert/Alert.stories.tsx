import * as React from "react";

import type { Meta, StoryObj } from "@storybook/react";

import { Alert } from "../Alert/Alert";

const meta: Meta<typeof Alert> = {
    component: Alert,
    tags: ["autodocs"],
};

export default meta;
type Story = StoryObj<typeof Alert>;

export const Warning: Story = {
    args: {
        title: "Attention needed",
        kind: "warning",
        children: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
        industry's standard dummy text ever since the 1500s, when`,
    },
    argTypes: {
        children: {
            table: {
                type: { summary: "string" },
                defaultValue: { summary: "undefined" },
            },
        },
        onCloseClick: {
            type: "function",
            table: {
                type: { summary: "() => void" },
                defaultValue: { summary: "undefined" },
            },
        },
    },
    render: props => <Alert {...props} />,
};
