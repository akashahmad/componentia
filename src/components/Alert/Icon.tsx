import * as React from "react";

import cx from "classnames";

import { WarningIcon } from "../icons/WarningIcon";
import { CheckmarkCircleIcon } from "../icons/CheckmarkCircleIcon";
import { CrossCircleIcon } from "../icons/CrossCircleIcon";
import { InformationAltIcon } from "../icons/InformationAltIcon";

import { kindPropType } from "./Alert";

export function AlertIcon({ kind }: { kind: kindPropType }) {
    const Icon = () => {
        if (kind === "success") return <CheckmarkCircleIcon />;
        else if (kind === "warning") return <WarningIcon />;
        else if (kind === "error") return <CrossCircleIcon />;
        else if (kind === "info") return <InformationAltIcon />;
    };

    return (
        <div
            className={cx(`inline-block rounded-md p-2 mr-[22px]`, {
                ["bg-blue-dark"]: kind === "info",
                ["bg-green"]: kind === "success",
                ["bg-red"]: kind === "error",
                ["bg-yellow"]: kind === "warning",
            })}
        >
            <Icon />
        </div>
    );
}
