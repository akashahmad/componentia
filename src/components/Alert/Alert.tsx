import * as React from "react";

import cx from "classnames";

import { AlertIcon } from "./Icon";
import { CloseIcon } from "../icons/CloseIcon";

export function Alert({ children, kind = "info", title = "Title Text", onCloseClick }: AlertProps) {
    return (
        <div
            className={cx("p-[35px] rounded-lg flex items-start border-l-[6px]", {
                ["border-blue-dark bg-blue-light-6"]: kind === "info",
                ["border-yellow bg-yellow-light-4"]: kind === "warning",
                ["border-green bg-green-light-6"]: kind === "success",
                ["border-red bg-red-light-6"]: kind === "error",
            })}
        >
            <AlertIcon kind={kind} />
            <div className={cx("w-full")}>
                <div className={cx("relative")}>
                    <h4
                        className={cx("leading-[26px] font-semibold text-lg mb-[14px]", {
                            ["text-info-text"]: kind === "info",
                            ["text-warning-text"]: kind === "warning",
                            ["text-success-text"]: kind === "success",
                            ["text-danger-text"]: kind === "error",
                        })}
                    >
                        {title}
                    </h4>
                    <div onClick={onCloseClick} className={cx("absolute top-0 right-0 w-5 cursor-pointer")}>
                        <CloseIcon />
                    </div>
                </div>
                <p
                    className={cx("text-base leading-6 font-normal", {
                        ["text-primary-text"]: kind === "info" || kind === "success",
                        ["text-warning-text-light"]: kind === "warning",
                        ["text-red-light"]: kind === "error",
                    })}
                >
                    {children}
                </p>
            </div>
        </div>
    );
}

export type kindPropType = "error" | "warning" | "info" | "success";
interface AlertProps {
    kind?: kindPropType;
    title: string;
    children: React.ReactNode;
    onCloseClick?: () => void;
}
