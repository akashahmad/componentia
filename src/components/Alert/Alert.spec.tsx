import * as React from "react";

import { render, screen } from "@testing-library/react";
import { Alert } from "./Alert";

test("Alert component is rendering properly", () => {
    const ALERT_TITLE = "TITLE";

    const { container } = render(
        <Alert title={ALERT_TITLE}>
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
            industrys standard dummy text ever since the 1500s, when
        </Alert>,
    );

    expect(screen.getByText(ALERT_TITLE)).not.toBeNull();

    expect(container).toMatchSnapshot();
});
