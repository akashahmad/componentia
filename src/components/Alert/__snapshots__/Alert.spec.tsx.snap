// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component is rendering properly 1`] = `
<div>
  <div
    class="p-[35px] rounded-lg flex items-start border-l-[6px] border-blue-dark bg-blue-light-6"
  >
    <div
      class="inline-block rounded-md p-2 mr-[22px] bg-blue-dark"
    >
      <svg
        fill="none"
        height="18"
        viewBox="0 0 18 18"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_727_29898)"
        >
          <path
            d="M9.0001 0.506348C4.30322 0.506348 0.506348 4.30322 0.506348 9.0001C0.506348 13.697 4.30322 17.522 9.0001 17.522C13.697 17.522 17.522 13.697 17.522 9.0001C17.522 4.30322 13.697 0.506348 9.0001 0.506348ZM9.0001 16.2563C5.00635 16.2563 1.77197 12.9938 1.77197 9.0001C1.77197 5.00635 5.00635 1.77197 9.0001 1.77197C12.9938 1.77197 16.2563 5.03447 16.2563 9.02822C16.2563 12.9938 12.9938 16.2563 9.0001 16.2563Z"
            fill="white"
          />
          <path
            d="M10.1249 7.65015H7.8749C7.5374 7.65015 7.22803 7.9314 7.22803 8.29702V13.922C7.22803 14.2595 7.50928 14.5689 7.8749 14.5689H10.1249C10.4624 14.5689 10.7718 14.2876 10.7718 13.922V8.29702C10.7718 7.9314 10.4624 7.65015 10.1249 7.65015ZM9.50615 13.2751H8.52178V8.91577H9.50615V13.2751Z"
            fill="white"
          />
          <path
            d="M8.9999 3.45947C8.04365 3.45947 7.22803 4.24697 7.22803 5.23135C7.22803 6.21572 8.01553 7.00322 8.9999 7.00322C9.98428 7.00322 10.7718 6.21572 10.7718 5.23135C10.7718 4.24697 9.95615 3.45947 8.9999 3.45947ZM8.9999 5.70947C8.71865 5.70947 8.49365 5.48447 8.49365 5.20322C8.49365 4.92197 8.71865 4.69697 8.9999 4.69697C9.28115 4.69697 9.50615 4.92197 9.50615 5.20322C9.50615 5.48447 9.28115 5.70947 8.9999 5.70947Z"
            fill="white"
          />
        </g>
        <defs>
          <clippath
            id="clip0_727_29898"
          >
            <rect
              fill="white"
              height="18"
              width="18"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="w-full"
    >
      <div
        class="relative"
      >
        <h4
          class="leading-[26px] font-semibold text-lg mb-[14px] text-info-text"
        >
          TITLE
        </h4>
        <div
          class="absolute top-0 right-0 w-5 cursor-pointer"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 10L18.625 2.375C18.9062 2.09375 18.9062 1.65625 18.625 1.375C18.3438 1.09375 17.9062 1.09375 17.625 1.375L10 9L2.375 1.375C2.09375 1.09375 1.65625 1.09375 1.375 1.375C1.09375 1.65625 1.09375 2.09375 1.375 2.375L9 10L1.375 17.625C1.09375 17.9062 1.09375 18.3438 1.375 18.625C1.5 18.75 1.6875 18.8438 1.875 18.8438C2.0625 18.8438 2.25 18.7812 2.375 18.625L10 11L17.625 18.625C17.75 18.75 17.9375 18.8438 18.125 18.8438C18.3125 18.8438 18.5 18.7812 18.625 18.625C18.9062 18.3438 18.9062 17.9062 18.625 17.625L11 10Z"
              fill="#374151"
            />
          </svg>
        </div>
      </div>
      <p
        class="text-base leading-6 font-normal text-primary-text"
      >
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when
      </p>
    </div>
  </div>
</div>
`;
